tests/test_pipeline_common.py

```diff
@@ -0,0 +1,507 @@
+
+    def test_pipeline_conversational(self):
+        self.run_task_tests(task="conversational")
+
+    @require_vision
+    @require_timm
+    @require_torch
+    def test_pipeline_depth_estimation(self):
+        self.run_task_tests(task="depth-estimation")
+
+    @require_pytesseract
+    @require_torch
+    @require_vision
+    def test_pipeline_document_question_answering(self):
+        self.run_task_tests(task="document-question-answering")
+
+    def test_pipeline_feature_extraction(self):
+        self.run_task_tests(task="feature-extraction")
+
+    def test_pipeline_fill_mask(self):
+        self.run_task_tests(task="fill-mask")
+
+    @require_torch_or_tf
+    @require_vision
+    def test_pipeline_image_classification(self):
+        self.run_task_tests(task="image-classification")
+
+    @require_vision
+    @require_timm
+    @require_torch
+    def test_pipeline_image_segmentation(self):
+        self.run_task_tests(task="image-segmentation")
+
+    @require_vision
+    def test_pipeline_image_to_text(self):
+        self.run_task_tests(task="image-to-text")
+
+    @require_vision
+    @require_timm
+    @require_torch
+    def test_pipeline_object_detection(self):
+        self.run_task_tests(task="object-detection")
+
+    def test_pipeline_question_answering(self):
+        self.run_task_tests(task="question-answering")
+
+    def test_pipeline_summarization(self):
+        self.run_task_tests(task="summarization")
+
+    def test_pipeline_table_question_answering(self):
+        self.run_task_tests(task="table-question-answering")
+
+    def test_pipeline_text2text_generation(self):
+        self.run_task_tests(task="text2text-generation")
+
+    def test_pipeline_text_classification(self):
+        self.run_task_tests(task="text-classification")
+
+    @require_torch_or_tf
+    def test_pipeline_text_generation(self):
+        self.run_task_tests(task="text-generation")
+
+    def test_pipeline_token_classification(self):
+        self.run_task_tests(task="token-classification")
+
+    def test_pipeline_translation(self):
+        self.run_task_tests(task="translation")
+
+    @require_torch_or_tf
+    @require_vision
+    @require_decord
+    def test_pipeline_video_classification(self):
+        self.run_task_tests(task="video-classification")
+
+    @require_torch
+    @require_vision
+    def test_pipeline_visual_question_answering(self):
+        self.run_task_tests(task="visual-question-answering")
+
+    def test_pipeline_zero_shot(self):
+        self.run_task_tests(task="zero-shot")
+
+    @require_vision
+    def test_pipeline_zero_shot_image_classification(self):
+        self.run_task_tests(task="zero-shot-image-classification")
+
+    @require_vision
+    @require_torch
+    def test_pipeline_zero_shot_object_detection(self):
+        self.run_task_tests(task="zero-shot-object-detection")
+
+
+def validate_test_components(test_case, task, model, tokenizer, processor):
```

Comment:
The goal of the PR is to move such logic in the model test files (or do you plan to do this in a followup)?

Reply:
Yes, a followup PR is much better, considering the amount of changes in this PR and for moving `validate_test_components` to model test files.